dora/core/server/master/src/main/java/alluxio/master/job/JournalMoveJobFactory.java

```diff
@@ -49,9 +51,10 @@ public Job<?> create() {
         mJobEntry.hasUser() ? Optional.of(mJobEntry.getUser()) : Optional.empty();
     UnderFileSystem ufs = mFs.getUfsManager().getOrAdd(new AlluxioURI(mJobEntry.getSrc()),
         UnderFileSystemConfiguration.defaults(Configuration.global()));
+    FilePredicate filePredicate = FilePredicate.create(mJobEntry.getFilter());
     Iterable<FileInfo> fileIterator =
         new UfsFileIterable(ufs, mJobEntry.getSrc(), user, mJobEntry.getPartialListing(),
-            FileInfo::isCompleted);
+            filePredicate.get());
```

Comment:
do you think we can combine create & get into one function?

Reply:
We could, but I think it would be more flexible if we have separate functions.